File: src/main/java/com/gs/service/BaseService.java
package com.gs.service;

import com.alibaba.fastjson.JSONObject;
import com.gs.model.City;
import com.gs.model.Type;
import org.springframework.stereotype.Service;

import java.util.List;

public interface BaseService {
    /**
     * 添加新的统计信息
     *
     * @param data
     * @return
     */
    JSONObject addNewData(JSONObject data);

    /**
     * 获取下一级别的类型
     *
     * @param data
     * @return
     */
    List<Type> getNextTypes(JSONObject data);

    /**
     * 获取类型的最高等级
     *
     * @param data
     * @return
     */
    Integer getMaxTypeLevel(JSONObject data);

    /**
     * 获取首页的力导向图的配置项
     * @param data
     * @return
     */
    JSONObject getTypeMap(JSONObject data);

    /**
     * 获取折线图的配置项
     * @param data
     * @return
     */
    JSONObject getLineBarOptions(JSONObject data);

    /**
     * 获取所有试卷名
     * @param data
     * @return
     */
    List<String> getAllExaminationName(JSONObject data);

    /**
     * 获取分析数据
     * @param data
     * @return
     */
    JSONObject getAnalyzeData(JSONObject data);
}

File: src/main/webapp/app/service/baseService.js
define(['app'],function (baseApp) {
    baseApp.factory('baseService', function ($timeout, $q, $http) {
        var factory = {};
        factory.addNewData = function(postDate){
            return factory.sendData("baseController/addNewData",postDate);
        }
        factory.getNextTypes = function(postDate){
            return factory.sendData("baseController/getNextTypes",postDate);
        }
        factory.getMaxTypeLevel = function(postDate){
            return factory.sendData("baseController/getMaxTypeLevel",postDate);
        }
        factory.getTypeMap = function(postDate){
            return factory.sendData("baseController/getTypeMap",postDate);
        }
        factory.getLineBarOptions = function(postDate){
            return factory.sendData("baseController/getLineBarOptions",postDate);
        }
        factory.getAllExaminationName = function(postDate){
            return factory.sendData("baseController/getAllExaminationName",postDate);
        }
        factory.addExaminationData = function(postDate){
            return factory.sendData("baseController/addExaminationData",postDate);
        }
        factory.getAnalyzeData = function(postDate){
            return factory.sendData("baseController/getAnalyzeData",postDate);
        }
        factory.sendData = function (url, params) {
            var defer = $q.defer();
            $http({
                url: url,
                method: 'POST',
                headers: {'Content-Type': 'application/json'},
                data: params,
            }).success(function (data) {
                defer.resolve(data);
            }).error(function (data, status, headers, config) {
                defer.reject(data);
            });
            return defer.promise;
        };
        return factory;
    });
});